checkov/kustomize/runner.py

```diff
@@ -171,19 +174,45 @@ def check_system_deps(self):
         # Ensure local system dependancies are available and of the correct version.
         # Returns framework names to skip if deps **fail** (ie, return None for a successful deps check).
         logging.info(f"Checking necessary system dependancies for {self.check_type} checks.")
-        try:
-            proc = subprocess.Popen([self.kustomize_command, 'version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)  # nosec
-            o, e = proc.communicate()
-            oString = str(o, 'utf-8')
-
-            if "Version:" in oString:
-                kustomizeVersionOutput = oString[oString.find('/') + 1: oString.find('G') - 1]
-                logging.info(f"Found working version of {self.check_type} dependancies: {kustomizeVersionOutput}")
-                return None
-            else:
-                return self.check_type
-        except Exception:
-            logging.info(f"Error running necessary tools to process {self.check_type} checks.")
+
+        if shutil.which(self.kubectl_command) != None:
+            try:
+                proc = subprocess.Popen([self.kubectl_command, 'version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)  # nosec
+                o, e = proc.communicate()
+                oString = str(o, 'utf-8')
+
+                if "Client Version:" in oString:
+                    kubectlVersionMajor = oString.split('\n')[0].split('Major:\"')[1].split('"')[0]
+                    kubectlVersionMinor = oString.split('\n')[0].split('Minor:\"')[1].split('"')[0]
+                    kubectlVersion = float(f"{kubectlVersionMajor}.{kubectlVersionMinor}")
+                    if kubectlVersion >= 1.14:
+                        logging.info(f"Found working version of {self.check_type} dependancy {self.kubectl_command}: {kubectlVersion}")
+                        self.templateRendererCommand = self.kubectl_command
+                        return None
+            
+            except Exception:
+                pass
+
+        elif shutil.which(self.kustomize_command) != None:
+    
+            try:
+                proc = subprocess.Popen([self.kustomize_command, 'version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)  # nosec
+                o, e = proc.communicate()
+                oString = str(o, 'utf-8')
+
+                if "Version:" in oString:
+                    kustomizeVersionOutput = oString[oString.find('/') + 1: oString.find('G') - 1]
+                    logging.info(f"Found working version of {self.check_type} dependancy {self.kustomize_command}: {kustomizeVersionOutput}")
+                    self.templateRendererCommand = self.kustomize_command
+                    return None
+                else:
+                    return self.check_type
+
+            except Exception:
+               pass
```

Comment:
add logging.debug with data on the exception

Reply:
Thanks 👍  Done!